prow/cmd/webhook-server/main.go

```diff
@@ -18,148 +18,176 @@ package main
 
 import (
 	"context"
-	"encoding/json"
 	"flag"
 	"fmt"
 	"io/ioutil"
 	"net/http"
 	"os"
 	"path/filepath"
-	"strings"
 
 	"github.com/sirupsen/logrus"
-	secretmanagerpb "google.golang.org/genproto/googleapis/cloud/secretmanager/v1"
 	"k8s.io/test-infra/experiment/clustersecretbackup/secretmanager"
-	"k8s.io/test-infra/prow/flagutil"
+	"k8s.io/test-infra/pkg/flagutil"
+	prowflagutil "k8s.io/test-infra/prow/flagutil"
 	"k8s.io/test-infra/prow/logrusutil"
+	ctrlruntimeclient "sigs.k8s.io/controller-runtime/pkg/client"
 )
 
 const (
-	secretID  = "prowjob-webhook-ca-cert"
-	caCert = "ca-cert"
-	caPrivKey = "ca-priv-key"
-	caBundle = "ca-bundle"
-	gcpSecretError = "failed to access secret version"
+	secretID                 = "prowjob-webhook-ca-cert"
+	configAlreadyExistsError = "already exists"
+	secretDirEmpty           = "secret dir was not created"
+	secretIncomplete         = "missing 1 or more secret files"
+	certFile                 = "certFile.pem"
+	privKeyFile              = "privKeyFile.pem"
+	caBundleFile             = "caBundle.pem"
 )
 
 type ClientInterface interface {
-	CreateSecret(ctx context.Context, secretID string) (*secretmanagerpb.Secret, error)
+	CreateSecret(ctx context.Context, secretID string) error
 	AddSecretVersion(ctx context.Context, secretName string, payload []byte) error
-	GetSecretValue(ctx context.Context, secretName, versionName string) ([]byte, error)
+	GetSecretValue(ctx context.Context, secretName string, versionName string) ([]byte, error)
+	CheckSecret(ctx context.Context, secretName string) (bool, error)
+}
+
+type options struct {
+	projectId      string
+	expiryInYears  int
+	dnsNames       []string
+	dryRun         bool
+	kubernetes     prowflagutil.KubernetesOptions
+	fileSystemPath string
+}
+
+func (o *options) Validate() error {
+	optionGroup := []flagutil.OptionGroup{&o.kubernetes}
+	if err := optionGroup[0].Validate(o.dryRun); err != nil {
+		return err
+	}
+	if o.expiryInYears < 0 {
+		return fmt.Errorf("invalid expiry years")
+	}
+	if len(o.projectId) > 0 && len(o.fileSystemPath) > 0 {
```

Comment:
super nit: In go it is more idiomatic to check if the string is populated like this: `o.fileSystemPath != ""` rather than checking the length.